packages/resilience/src/circuit-breaker.ts

```diff
@@ -0,0 +1,147 @@
+import CircuitBreaker from 'opossum';
+// eslint-disable-next-line import/named
+import { AxiosError } from 'axios';
+import {
+  Context,
+  HttpMiddlewareContext,
+  Middleware,
+  MiddlewareIn
+} from './middleware';
+
+type ErrorFilter = (err) => boolean;
+type KeyBuilder<ContextT extends Context> = (context: ContextT) => string;
+
+function httpErrorFilter(error: AxiosError): boolean {
+  if (
+    error.response?.status &&
+    error.response.status.toString().startsWith('4')
+  ) {
+    return true;
+  }
+  return false;
+}
+
+function isAxiosError(err: AxiosError | Error): err is AxiosError {
```

Comment:
[pp] I think we could make our lives easier and just have one CB for HTTP and XSUAA and filter for both just the 400 axios errors out. A wrong config with respect to client credentials is unlikely in a real scenario where this comes from the service binding.
If you want to keep them both also fine.

Reply:
I simplified this, so we now have a common function for sb for xsuaa and http